server/src/main/java/org/apache/druid/server/QueryResultPusher.java

```diff
@@ -223,6 +242,10 @@ private Response handleDruidException(ResultsWriter resultsWriter, DruidExceptio
         // also throwing the exception body into the response to make it easier for the client to choke if it manages
         // to parse a meaningful object out, but that's potentially an API change so we leave that as an exercise for
         // the future.
+
+        if (includeTrailerHeader) {
+          trailerFields.put(QueryResource.ERROR_MESSAGE_TRAILER_HEADER, e.getMessage());
```

Comment:
IMO we should have one trailer to indicate whether the response was complete or not. And the error message can be populated in another trailer.

Reply:
Seems like good idea. I will add one. Thanks

Reply:
done